extensions/amp-form/0.1/amp-form.js

```diff
@@ -147,12 +160,36 @@ export class AmpForm {
     /** @const @private {!./form-validators.FormValidator} */
     this.validator_ = getFormValidator(this.form_);
 
-    this.actions_.installActionHandler(
-        this.form_, this.actionHandler_.bind(this));
+    // Wait for amp-selector to finish loading if the current form is using it.
+    this.whenDependenciesReady_().then(() => {
+      this.actions_.installActionHandler(
+          this.form_, this.actionHandler_.bind(this));
+    });
     this.installEventHandlers_();
   }
 
   /**
+   * Returns a promise that will be resolved when all dependencies used inside the form
+   * tag are loaded (e.g. amp-selector) or 500ms timeout - whichever is first.
+   * @return {!Promise}
+   * @private
+   */
+  whenDependenciesReady_() {
+    const usedDeps = this.form_.querySelectorAll(EXTERNAL_DEPS.join(','));
+    const depsDict = {};
+    const depsPromises = [];
+    for (let i = usedDeps.length - 1; i >= 0; i--) {
```

Comment:
I guess now we only have one dependency `getElementsByTagName` is fine. Not sure if `querySelectorAll` would be faster than multiple `getElementsByTagName` in the future.

Will update this to use `getElementsByTagName`.